searx/searx
# Processors

## What does this PR do?

Create one "processor" (*) per one class per engine type:
* offline
* online
  * online_currency
  * online_dictionnary

Each processor has two methods:
```python
   def get_params(self, search_query, category):
      pass

   def search(self, query, params, result_container, start_time, timeout_limit):
      pass  
```

[online_currency](https://github.com/searx/searx/blob/f938072b5451cdcbea506707c4d4af1ccbcc333c/searx/search/processors/online_currency.py) and [online_dictionnary](https://github.com/searx/searx/blob/f938072b5451cdcbea506707c4d4af1ccbcc333c/searx/search/processors/online_dictionary.py) checks if the query match the excepted format ( equivalent of #2159 ).

There is [one processor per engine](https://github.com/searx/searx/blob/f938072b5451cdcbea506707c4d4af1ccbcc333c/searx/search/processors/__init__.py).

## Why is this change important?

Once merged, it will be easier to implement
* https://github.com/searx/searx/issues/1559
* https://github.com/searx/searx/pull/477
* and most probably https://github.com/searx/searx/issues/2067

## How to test this PR locally?

* activate `` and `` engines, then search for `en-es house`
* search for `10 eur in jpy`
* make a query using the `time_range` parameter
* make a query using the `language` parameter
* activate the `command` engine with the following configuration, and make sure it is working:
```yaml
  - name: locate
    engine: command
    command: ['locate', '--existing', '--ignore-case', '{{QUERY}}']
    shortcut: locate
    tokens: []
    disabled: False
    delimiter:
        chars: ' '
        keys: ['line']
```

## Author's checklist



## Related issues
